src/launcher/components/AppIcon.jsx

```diff
@@ -38,34 +38,26 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import semver from 'semver';
 
-function renderAlert(altText) {
-    return (
-        <div>
-            <span className="alert-icon-bg" />
-            <span className="mdi mdi-alert" title={altText} />
-        </div>
-    );
-}
+import checkAppCompatibility from '../util/checkAppCompatibility';
+
+const renderAlert = altText => (
+    <div>
+        <span className="alert-icon-bg" />
+        <span className="mdi mdi-alert" title={altText} />
+    </div>
+);
 
-function renderNotice(app) {
-    if (!app.engineVersion) {
-        return renderAlert(
-            'The app does not specify which nRF Connect version(s) ' +
-                'it supports'
-        );
-    }
-    if (!app.isSupportedEngine) {
-        return renderAlert(
-            `The app only supports nRF Connect ${app.engineVersion}, ` +
-                'which does not match your currently installed version'
-        );
-    }
-    return null;
-}
+const renderNotice = app => {
+    const installed = !!app.currentVersion;
+    const appCompatibility = checkAppCompatibility(app);
+
+    return !installed || appCompatibility.isCompatible
```

Comment:
[minor] Might be more readable to replace `installed` with `notInstalled` to prevent some negations. 
```js
const notInstalled = !app.currentVersion
...

return notInstalled || appCompatibility.isCompatible
...
```

Reply:
Good idea. I see there are also some more things in `AppIcon.jsx` that I would like to change, e.g. the mixture of styles in JS and in SCSS. I have also some other cleanups in the pipeline to do after this PR is merged, so I will add these to the list.